py/objdeque.c

```diff
@@ -135,6 +183,53 @@ STATIC mp_obj_t deque_popleft(mp_obj_t self_in) {
 }
 STATIC MP_DEFINE_CONST_FUN_OBJ_1(deque_popleft_obj, deque_popleft);
 
+STATIC mp_obj_t deque_pop(mp_obj_t self_in) {
+    mp_obj_deque_t *self = MP_OBJ_TO_PTR(self_in);
+
+    if (self->i_get == self->i_put) {
+        mp_raise_msg(&mp_type_IndexError, MP_ERROR_TEXT("empty"));
+    }
+
+    if (self->i_put == 0) {
+        self->i_put = self->alloc - 1;
+    } else {
+        self->i_put--;
+    }
+
+    mp_obj_t ret = self->items[self->i_put];
+
+    return ret;
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_1(deque_pop_obj, deque_pop);
+
+STATIC mp_obj_t deque_getiter(mp_obj_t o_in, mp_obj_iter_buf_t *iter_buf) {
+    mp_obj_deque_t *deque = MP_OBJ_TO_PTR(o_in);
+    return mp_obj_new_deque_iterator(o_in, deque->i_get, iter_buf);
+}
+
+STATIC size_t deque_get_index(mp_obj_deque_t *self, mp_obj_t index) {
```

Comment:
Although this would be inlined by the compiler, the general style in MicroPython is to write this directly in the code, ie not to have helper functions that are only called from one location.

Reply:
It ended up being a lot less code than when I started the refactor, that makes sense.